5-April-22/upgrade.md

```diff
@@ -0,0 +1,140 @@
+# Juno April 5, 2022 Chain Halt Upgrade
+This upgrade is effectively a hard fork with the same chain-id. This means we'll need to delete all previous data and start from a new genesis.
+
+NOTE: This assumes you've already ran through setting up a juno node here: https://docs.junonetwork.io/validators/getting-setup
+
+## Performing the  Upgrade
+
+### 1. Stop the node
+```sh
+sudo systemctl stop junod
+```
+
+### 2. Backup priv_validator_key.json
+This is only relevant if you're upgrading your validator. Your priv_validator_key.json is how your validator is identified. If you haven't backed it up already, **DO SO NOW**.
+
+An example method for doing so is as follows, which will copy your validator key to the home dir:
+```sh
+cd ~
+cp ~/.juno/config/priv_validator_key.json .
+```
+
+*NOTE*: keeping your copy of the validator key on the same machine is NOT sufficient. At a bare minimum it should be backed up locally so you always have access to it.
+
+### 3. Backup priv_validator_state.json
+It's worth backing up your `priv_validator_state.json` in case this upgrade fails and an alternative path must be taken. 
+```sh
+cd ~
+cp ~/.juno/data/priv_validator_state.json .
+```
+
+### 4. Purge previous chain state and addrbook.json
+Because we'll be starting from a new genesis, the previous data is no longer necessary.
+```sh
+junod unsafe-reset-all
+```
+
+### 5. Purge current peers and seeds from config.toml
+This is done to ensure all peers are clean when moving forward. The addrbook.json file was already purged in the previous step.
+```sh
+sed -i.bak -e "s/^persistent_peers *=.*/persistent_peers = \"\"/" ~/.juno/config/config.toml
+sed -i.bak -e "s/^seeds *=.*/seeds = \"\"/" ~/.juno/config/config.toml
+```
+
+### 6. Add seeds and peers to config.toml
+These are all verified to be using the new genesis file and binary.
+```sh
+SEEDS=
+PEERS=
+sed -i.bak -e "s/^seeds *=.*/seeds = \"$SEEDS\"/" ~/.juno/config/config.toml
+sed -i.bak -e "s/^persistent_peers *=.*/persistent_peers = \"$PEERS\"/" ~/.juno/config/config.toml
+```
+
+### 7. Download and install the new binary
+
+#### 7a. Install junod v3.0.0
+This new binary includes the Lupercalia security upgrade.
+```sh
+git clone https://github.com/CosmosContracts/juno
+cd juno
+git fetch
+git checkout v3.0.0
```

Comment:
`git checkout v2.3.0` I believe (for now)

Reply:
No it's good.